controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -539,6 +545,26 @@ func getCurrentImage(host *metal3v1alpha1.BareMetalHost) *metal3v1alpha1.Image {
 	return nil
 }
 
+// actionDetached() detaches the host from the Provisioner
+func (r *BareMetalHostReconciler) actionDetached(prov provisioner.Provisioner, info *reconcileInfo) actionResult {
```

Comment:
nit: I think of this as the counterpart to `registerHost()` more than an equivalent to the other methods starting with `action`, which are all called from the state machines `handle` methods. So maybe `detachHost()` would be a good name.

Reply:
Thanks, renamed.